src/SFML/Window/Win32/WglContext.cpp

```diff
@@ -276,6 +276,31 @@ void WglContext::setVerticalSyncEnabled(bool enabled)
 ////////////////////////////////////////////////////////////
 int WglContext::selectBestPixelFormat(HDC deviceContext, unsigned int bitsPerPixel, const ContextSettings& settings, bool pbuffer)
 {
+    struct PixelFormatCacheEntry
+    {
+        unsigned int bitsPerPixel;
+        unsigned int depthBits;
+        unsigned int stencilBits;
+        unsigned int antialiasingLevel;
+        bool         pbuffer;
+        int          bestFormat;
+    };
+
+    static std::mutex                         cacheMutex;
+    static std::vector<PixelFormatCacheEntry> pixelFormatCache;
```

Comment:
Just random thinking, didn't consider this last time with the original PR: Shouldn't there be one cache per device context instead of one global one? Haven't tried it, but might this screw up things in case displays change at runtime? Let's say switching between a HDR screen and a non-HDR screen?

Reply:
Pixel formats are specific to the GPU that Windows chooses to use for rendering the application window. If you run multiple screens, some being HDR and others SDR you can move the window between them and the OpenGL/GDI/Compositor will take care of the magic for you. Since we can't even select the GPU to render on in OpenGL, having one cache should be good enough.